truss/templates/server/model_wrapper.py

```diff
@@ -291,6 +293,8 @@ async def write_response_to_queue(
     ):
         with tracing.section_as_event(span, "write_response_to_queue"):
             try:
+                # Special case for writer: the triton client checks for canellations
+                # in each iteration.
                 async for chunk in generator:
```

Comment:
Alternatively we could check here, and use `generator.aclose()` which would raise `GeneratorExit` inside the producer loop.

The downside of this is that it is a bit more implicit.

Reply:
what are we doing differently here?